application/src/main/java/org/openhds/mobile/utilities/OdkCollectHelper.java

```diff
@@ -149,4 +159,101 @@ public static void setStatusComplete(ContentResolver resolver, Uri uri) {
         resolver.update(uri, cv, null, null);
 
     }
+
+    private static String makePlaceholders(int len) {
+        if (len < 1) {
+            // It will lead to an invalid query anyway ..
+            throw new RuntimeException("No placeholders");
+        } else {
+            StringBuilder sb = new StringBuilder(len * 2 - 1);
+            sb.append("?");
+            for (int i = 1; i < len; i++) {
+                sb.append(",?");
+            }
+            return sb.toString();
+        }
+    }
+
+    public static List<FormInstance> getFormInstancesByPath(ContentResolver resolver, Collection<String> ids) {
```

Comment:
Just an idea: usually when names become long, I try to lean on namespacing and type information to clarify the method without long names. For example, here the signature clearly states it returns a list of form instances so the method names could be simply (get/find)ByPath(s)?